src/middlewares/likeDislikeCheck.js

```diff
@@ -0,0 +1,60 @@
+import models from '../api/models';
+import statusCodes from '../helpers/constants/status.codes';
+
+/**
+ * @class checkLikesDislikes
+ * @description a class to check if a user has already like or disliked a comment
+ */
+export default class checkLikesDislikes {
+  /**
+   * Verify if the user has already liked the comment
+   * @param {Object} req - Request
+   * @param {Object} res  - Response
+   * @param {Function} next -Next
+   * @returns {Object} The response object
+   */
+  static async liked(req, res, next) {
+    const { commentId } = req.params;
+    const { id } = req.user.user;
+    const hasLiked = await models.LikeDislike.findAll({
+      where: {
+        commentId,
+        userId: id,
+        likes: 1
+      }
+    });
+    // If the user has already liked that comment
+    if (hasLiked[0]) {
+      return res.status(statusCodes.OK).json({
+        message: 'You have already liked this comment!'
+      });
+    }
+    next();
+  }
+
+  /**
+   * Verify if the user has already disliked the comment
+   * @param {Object} req - Request
+   * @param {Object} res  - Response
+   * @param {Function} next -Next
+   * @returns {Object} The response object
+   */
+  static async disliked(req, res, next) {
```

Comment:
and to avoid the duplication ``maintainability``, I think you can separate these middlewares into two files

Reply:
The maintainability is fine. Can you explain me more what exactly you want me to do